fix permissions

ann26 opened the pull request:
fix #646 

Added edit and delete button on sponsor details page. Staff, project owner, project changelog manager and the author of the sponsor can update/delete it.
![screenshot from 2017-12-12 13-38-55](https://user-images.githubusercontent.com/26101337/33870732-d47166b2-df41-11e7-9add-1aee570c9463.png)

For other logged in users
![screenshot from 2017-12-12 13-43-49](https://user-images.githubusercontent.com/26101337/33870864-844b4e72-df42-11e7-8189-f10f4dc9fada.png)

For non-logged in users
![screenshot from 2017-12-12 13-20-46](https://user-images.githubusercontent.com/26101337/33870759-fa365344-df41-11e7-86ea-50efabc22e4a.png)

timlinux commented:
Thanks. Can you remove changelog manager permissions and perhaps consider adding a similar field but using a new 'sponsorship manager' project property that specifies who is allowed to edit sponsors?

ann26 commented:
ok @timlinux will do

ann26 commented:
- [x] Added sponsorship manager to project's model
- [x] Added sponsorship manager to project form
- [x] Sponsorship manager can approve, edit/delete sponsor and sponsorship period

![screenshot from 2017-12-20 11-22-26](https://user-images.githubusercontent.com/26101337/34196886-965756a6-e596-11e7-85e7-e87618ec55bb.png)

timlinux commented:
Brilliant LGTM thanks @ann26
